vm/src/builtins/genericalias.rs

```diff
@@ -0,0 +1,136 @@
+use crate::builtins::PyStr;
+use crate::builtins::PyTuple;
+use crate::builtins::PyTypeRef;
+use crate::common::hash;
+use std::fmt;
+
+use crate::slots::{Hashable, SlotConstructor};
+use crate::{
+    IdProtocol, PyClassImpl, PyContext, PyObjectRef, PyRef, PyResult, PyValue, VirtualMachine,
+};
+
+#[pyclass(module = false, name = "types.GenericAlias")]
```

Comment:
```suggestion
#[pyclass(module = "types", name = "GenericAlias")]
```

I also wish you try to use `#[pymodule]` for new modules.

Reply:
I fixed what you suggested. But I wonder it needs to declare a new `module`. There is already a `types` module written in Python (`types.py`). For example, `cmath` module was declared through `#[pymodule]` because there is no `cmath.py` Python file.

Reply:
pymodules can be merged. See `stdlib/time.rs` for other examples.
```rust
types::extend_module(vm, &module);
```

Reply:
I think I misunderstood something. You defined this type in `builtins::genericalias` like builtin types. But this is not a bulitin type but defined in `_testcapi` module.

Reply:
This is weird. I also found `_GenericAlias` is defined in `typing.py`. How this is working?

Reply:
As I understood:
- `typing._GenericAlias` and `types.GenericAlias` are different
- CPython doesn't expose `types.GenericAlias` builtin direct type but the `types.GenericAlias` is exposed by `type(list[int])` indirectly. About this, I didn't understand its context well 😞 . `typing._GenericAlias` has a similar naming but independent class which acts differently.

Reply:
Thank you! I got it.
```
Python 3.9.5 (default, May 26 2021, 13:13:06) 
[Clang 12.0.5 (clang-1205.0.22.9)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import _testcapi
>>> _testcapi.GenericAlias
<class 'GenericAlias'>
>>> _testcapi.GenericAlias.__qualname__
'GenericAlias'
>>> _testcapi.GenericAlias.__module__
'builtins'
```